src/java.net.http/share/classes/java/net/http/HttpClient.java

```diff
@@ -730,4 +745,126 @@ public enum Redirect {
     public WebSocket.Builder newWebSocketBuilder() {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Initiates an orderly shutdown in which previously submitted
+     * operations are completed, but no new request will be accepted.
+     * Invocation has no additional effect if already shut down.
+     *
+     * <p>This method does not wait for previously submitted request
+     * to complete execution.  Use {@link #awaitTermination awaitTermination}
+     * to do that.
+     *
+     * @implSpec
+     * The default implementation of this method does nothing. Subclasses should
+     * override this method to implement the appropriate behavior.
+     * See the API Note on {@linkplain ##closing closing}.
+     *
+     * @since 21
+     */
+    public void shutdown() { }
+
+    /**
+     * Blocks until all operations have completed execution after a shutdown
+     * request, or the timeout occurs, or the current thread is
+     * interrupted, whichever happens first.
+     *
+     * @implSpec
+     * The default implementation of this method does nothing and returns true.
```

Comment:
If that where the case, then `close()` would never return when none of these methods are overridden (case of existing subclasses).  `isTerminated` stipulates that it will never return `true` if shutdown has not been called, which is borrowed from `ExecutorService` specification, and maintaining a state in this super class is not really a good idea. The choice was being throwing an exception (that would make close throw too), or return `true`, which would allow close() to trivially exit if not implemented. All in all I believe I prefer to have `awaitTermination` returning `true` by default.